connect/runtime/src/test/java/org/apache/kafka/connect/util/clusters/EmbeddedConnectCluster.java

```diff
@@ -559,6 +560,31 @@ public String endpointForResource(String resource) {
         return url + resource;
     }
 
+    /**
+     * Get the full URL of the endpoint that corresponds to the given REST resource using a worker
+     * that is not running any tasks or connector instance for the connectorName provided in the arguments
+     *
+     * @param resource the resource under the worker's admin endpoint
+     * @param connectorName the name of the connector
+     * @return the admin endpoint URL
+     * @throws ConnectException if no REST endpoint is available
+     */
+    public String endpointForWorkerRunningNoResourceForConnector(String resource, String connectorName) {
```

Comment:
What do you think about returning `Optional<String>` here (if every worker has at least one instance for the given connector) rather than throwing an exception?

Reply:
@rhauch  I think it's a good idea but I saw that org.apache.kafka.connect.util.clusters.EmbeddedConnectCluster#endpointForResource method is using the same pattern and if I change it then I will have to change a lot of classes not related to this PR and ultimately in the test I will have to check for Optional.present and do an assert.  The situation for not finding a worker is rare and doing empty check and assert in all places will be a lot of copy/paste whereas throwing an exception here is centralizing this rare situation. Do you think it's ok to leave it as is?

Reply:
Well, the `EmbeddedConnectCluster` is technically not an official public API for Connect, but lots of connector projects use it (as intended), meaning we have to be very clear about backward compatibility. It's probably fine to keep the same pattern.

Reply:
Good idea, I renamed the method.